Lib/Protocols/IdHL7.pas

```diff
@@ -1582,24 +1843,35 @@ function TIdHL7.GetMessage(var VMsg: String): TObject;
     end;
 end;
 
-procedure TIdHL7.SendReply(AMsgHnd: TObject; AReply: String);
+Procedure TIdHL7.SendReply(AMsgHnd: pointer; AReply: AnsiString);
 var
-  qm: TQueuedMessage;
+  qm: TIdQueuedMessage;
 begin
   assert(Assigned(self));
   assert(Assigned(AMsgHnd));
-  assert(AReply <> '', 'Attempt to send an empty reply'); {do not localize}
+//  assert(AReply <> '', 'Attempt to send an empty reply'); {do not localize}
+  Assert(Length(AReply) > 0, 'Attempt to send an empty reply'); {do not localize}
   assert(Assigned(FLock));
   FLock.Enter;
   try
-    qm := AMsgHnd as TQueuedMessage;
+    qm := TObject(AMsgHnd) As TIdQueuedMessage;
     qm.FReply := AReply;
+    qm.FEvent.SetEvent;
     qm._Release;
     FHndMsgQueue.Delete(FHndMsgQueue.IndexOf(AMsgHnd));
   finally
     FLock.Leave;
     end;
-  qm.FEvent.SetEvent;
+ //qm.FEvent.SetEvent;
 end;
 
+Function TIdHL7ClientThread.TimedOut: Boolean;
+Var
+  lGap : TDateTime;
+Begin
+  lGap := (now - FLastTraffic) * 24 * 60 * 60 * 1000;

```

Comment:
Prefer to use `DateUtils` functions, when available, to manipulate `TDateTime` values

Reply:
Made change regarding this as far as possible for the logic
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9